lib/db_connection/ownership/owner.ex

```diff
@@ -65,13 +65,21 @@ defmodule DBConnection.Ownership.Owner do
     {:ok, state}
   end
 
-  def handle_info({:DOWN, mon, _, pid, reason}, %{client: {_, mon}} = state) do
-    message = "client #{inspect pid} exited: " <> Exception.format_exit(reason)
+  def handle_info({:DOWN, mon, _, pid, reason},
+                  %{client: {_, _, mon}} = state) do
+    message = "client #{inspect pid} exited with: " <> Exception.format_exit(reason)
     disconnect(message, state)
   end
 
-  def handle_info({:DOWN, ref, _, pid, reason}, %{owner_ref: ref} = state) do
-    message = "owner #{inspect pid} exited: " <> Exception.format_exit(reason)
+  def handle_info({:DOWN, ref, _, pid, reason},
+                  %{owner_ref: ref, client: nil} = state) do
```

Comment:
since we detect no client here can prevent duplicate logic in `down/2` and just do the checkin instead of creating a message only not to use it?